ta4j-examples/src/main/java/ta4jexamples/backtesting/SimpleMovingAverageRangeBacktesting.java

```diff
@@ -0,0 +1,45 @@
+package ta4jexamples.backtesting;
+
+import org.ta4j.core.*;
+import org.ta4j.core.indicators.SMAIndicator;
+import org.ta4j.core.indicators.helpers.ClosePriceIndicator;
+import org.ta4j.core.num.PrecisionNum;
+import org.ta4j.core.trading.rules.OverIndicatorRule;
+import org.ta4j.core.trading.rules.UnderIndicatorRule;
+import ta4jexamples.loaders.CsvBarsLoader;
+
+public class SimpleMovingAverageRangeBacktesting {
+
+    public static void main(String[] args) throws InterruptedException {
+        TimeSeries series = CsvBarsLoader.loadAppleIncSeries();
+
+        Backtesting backtesting = new Backtesting(series);
+
+        int start = 3;
+        int stop = 50;
+        int step = 5;
+
+        for(int i=start; i<=stop; i += step) {
+            Strategy strategy = new BaseStrategy(
+                    "Sma("+ i +")",
+                    createEntryRule(series, i),
+                    createExitRule(series, i)
+            );
+            backtesting.addStrategy(strategy);
+        }
+        backtesting.calculate(Order.OrderType.BUY, PrecisionNum.valueOf(50));
+        backtesting.printBacktestingResults(true);
+    }
+
+    private static Rule createEntryRule(TimeSeries series, int barCount) {
+        Indicator closePrice = new ClosePriceIndicator(series);
```

Comment:
Should be 'Indicator<Num>' instead of Indicator otherwise we have unchecked assignment: 'org.ta4j.core.Indicator' to 'org.ta4j.core.Indicator<org.ta4j.core.num.Num>

Reply:
What are you suggesting here?

Reply:
Sorry, my fault. Should be 'Indicator<Num>' instead of 'Indicator'

Reply:
Ok, I see. It was not my fault, but github is removing my brackets in comments. So I want to say Indicator has missing generic param